File: app/src/main/java/co/kubo/indiesco/activities/NotificacionIDTokenService.java
package co.kubo.indiesco.activities;

import android.util.Log;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

/**
 * Created by estacion on 21/02/18.
 */

class NotificacionIDTokenService extends FirebaseInstanceIdService {

    private static final String TAG = "FIREBASE TOKEN";

    @Override
    public void onTokenRefresh() {
        //super.onTokenRefresh();
        Log.d(TAG, "Solicitando token");
        String token = FirebaseInstanceId.getInstance().getToken();
    }//public void onTokenRefresh

    private void sendRegistrationToServer(String token){
        Log.d(TAG, token);
    }//private void sendRegistrationToServer

}//public class NotificacionIDTokenService

File: app/src/main/java/co/kubo/indiesco/fcm/NotificacionIDTokenService.kt
package co.kubo.indiesco.fcm

import android.util.Log
import co.kubo.indiesco.utils.SharedPreferenceManager
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService

/**
 * Created by estacion on 1/06/18.
 */
class NotificacionIDTokenService : FirebaseInstanceIdService() {

    private val TAG = "FIREBASE_TOKEN"
    val sharedPreferenceManager = SharedPreferenceManager()

    override fun onTokenRefresh() {
        //super.onTokenRefresh()
        Log.d(TAG, "Solicitando token")
        val token = FirebaseInstanceId.getInstance().token
        sharedPreferenceManager.setFirebaseToken(applicationContext, token)
        sendRegistrationToServer(token!!)
    }

    private fun sendRegistrationToServer(token: String) {
        Log.d(TAG, token)
    }
}